mmgen/models/architectures/sngan_proj/generator_discriminator.py

```diff
@@ -0,0 +1,296 @@
+from copy import deepcopy
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import ConvModule, build_activation_layer, xavier_init
+from mmcv.runner import load_checkpoint
+from mmcv.utils import build_from_cfg
+from torch.nn.utils import spectral_norm
+
+from mmgen.models.builder import MODULES
+from mmgen.utils import check_dist_init
+from mmgen.utils.logger import get_root_logger
+from ..common import get_module_device
+
+
+@MODULES.register_module()
+class SNGANGenerator(nn.Module):
+
+    # TODO: add doc here
+    _defualt_channels_cfg = {
+        32: [1, 1, 1],
+        64: [16, 8, 4, 2],
+        128: [16, 16, 8, 8, 4, 2]
+    }
+
+    _default_blocks_cfg = dict(type='SNGANGenResBlock')
+
+    def __init__(self,
+                 output_scale,
+                 noise_size,
+                 num_classes=0,
+                 out_channels=3,
+                 base_channels=128,
+                 input_scale=4,
+                 blocks_cfg=None,
```

Comment:
I may suggest that you can directly use `blocks_cfg=dict(type='SNGANGenResBlock')`. Then, we can remove `_default_blocks_cfg` from this class. Because the default config only contains little information.